src/tsickle.ts

```diff
@@ -203,87 +202,62 @@ class ClosureRewriter extends Rewriter {
-            if (!paramData.types[i]) {
-              paramData.types[i] = new Set<string>();
-            }
-            paramData.types[i].add(newTag.type);
+        for (let {tagName, parameterName, text} of jsDoc) {
+          if (tagName === 'param' && parameterName === newTag.parameterName) {
+            newTag.text = text;
+            break;
           }
         }
-      }
-      // If this method is not overloaded, we set the list of names to those in the only declaration
-      if (!isOverloaded) {
-        paramNamesToReturn = fnDecl.parameters.map(getParameterName);
+        if (!paramTags[i]) paramTags.push([]);
+        paramTags[i].push(newTag);
       }
 
       // Return type.
       if (!isConstructor) {
         let retType = typeChecker.getReturnTypeOfSignature(sig);
         let retTypeString: string = this.typeToClosure(fnDecl, retType);
-        if (!isOverloaded) {
-          // TODO: Consider adding text from existing JSDoc into overloads
-          let returnDoc: string|undefined;
-          for (let {tagName, text} of jsDoc) {
-            if (tagName === 'return') {
-              returnDoc = text;
-              break;
-            }
-          }
-          newDoc.push({
-            tagName: 'return',
-            type: retTypeString,
-            text: returnDoc,
-          });
-        } else {
-          if (retTypeString) {
-            paramData.returns.add(retTypeString);
+        let returnDoc: string|undefined;
+        for (let {tagName, text} of jsDoc) {
+          if (tagName === 'return') {
+            returnDoc = text;
+            break;
           }
         }
+        returnTags.push({
+          tagName: 'return',
+          type: retTypeString,
+          text: returnDoc,
+        });
       }
     }
-    if (isOverloaded) {
-      // Build actual JSDoc tags for the merged param names/types and return types
-      for (let i = 0; i < maxCount; i++) {
-        paramNamesToReturn.push(Array.from(paramData.names[i].values()).join('_or_'));
-        if (i >= minCount) {
-          paramNamesToReturn[i] = 'opt_' + paramNamesToReturn[i];
-        }
-        let concatenatedTypes = Array.from(paramData.types[i].values()).join('|');
-        newDoc.push({
-          tagName: 'param',
-          parameterName: paramNamesToReturn[i],
-          type: concatenatedTypes,
-        });
+
+    // Merge the JSDoc tags for each overloaded parameter.
+    for (let i = 0; i < maxArgsCount; i++) {
+      let paramTag = jsdoc.merge(paramTags[i]);
+      // If any overload marks this param optional, mark it optional in the
+      // merged output.
+      const optional = paramTags[i].find(t => t.optional === true) !== undefined;
+      if (optional || i >= minArgsCount) {
+        // TODO(evanm): this successfully renames the parameter in the JSDoc,
+        // but we also need to rename the parameter in the emitted function.
+        // paramTag.parameterName = 'opt_' + paramTag.parameterName;
```

Comment:
I think `opt_` is just a coding convention, not load bearing (and also no longer recommended by style guide). the `=` is the core part. So unless you care about the style, you don't need that here.